yamale/tests/test_functional.py

```diff
@@ -154,15 +191,67 @@ def test_bad_regexes():
     assert count_exception_lines(regexes['schema'], regexes['bad']) == 9
 
 
+def test_bad_include_validator():
+    # TODO use match
+    exp = ["key1: 'a_string' is not a int."]
+    match_exception_lines(include_validator['schema'],
+                          include_validator['bad'],
+                          exp)
+    #assert count_exception_lines(include_validator['schema'], include_validator['bad']) == 3
+
+
 def test_bad_schema():
     with pytest.raises(SyntaxError) as excinfo:
         yamale.make_schema(get_fixture('bad_schema.yaml'))
     assert 'fixtures/bad_schema.yaml' in str(excinfo.value)
 
 
-def count_exception_lines(schema, data):
+def test_list_is_not_a_map():
+    exp = [": '[1, 2]' is not a map"]
+    match_exception_lines(strict_map['schema'],
+                          strict_list['good'],
+                          exp)
+
+
+def test_bad_strict_map():
+    exp = ['extra: Unexpected element']
+    match_exception_lines(strict_map['schema'],
+                          strict_map['bad'],
+                          exp,
+                          strict=True)
+
+
+def test_bad_strict_list():
+    exp = ['2: Unexpected element']
+    match_exception_lines(strict_list['schema'],
+                          strict_list['bad'],
+                          exp,
+                          strict=True)
+
+def test_bad_mixed_strict_map():
+    exp = ['field3.extra: Unexpected element']
+    match_exception_lines(mixed_strict_map['schema'],
+                          mixed_strict_map['bad'],
+                          exp,
+                          strict=False)
+
+
+def match_exception_lines(schema, data, expected, strict=False):
```

Comment:
I'd rather see this written as the following:
```
def match_exception_lines(schema, data, expected, strict=False):
    with pytest.raises(ValueError) as e:
        assert yamale.validate(schema, data, strict)

    message = str(e.value)
    # only match the actual error message and remove the leading \t
    got = set(s.lstrip() for s in message.split('\n') if s.startswith('\t'))
    expected = set(expected)
    assert got == expected
```

Or if you change the `expected` parameter to a string, you could do this:
```
def match_exception_lines(schema, data, expected, strict=False):
    with pytest.raises(ValueError, match=re.escape(expected)):
        assert yamale.validate(schema, data, strict)
```

Reply:
Updated to use the first version you suggested. I didn't know the assert would give such a useful error message for set comparison. Nice!